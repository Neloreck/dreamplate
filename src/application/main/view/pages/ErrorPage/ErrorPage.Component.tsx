import { Bind, Consume } from "dreamstate";
import { PureComponent, ReactNode } from "react";
import { WithSheet } from "react-jss";

// Lib.
import { Styled, Wrapped } from "@Lib/decorators";

// Data.
import { IRouterContext, routerContextManager } from "@Main/data/store";

// View.
import { IMainHeaderInjectedProps, MainHeader } from "@Main/view/components/MainHeader";
import { GlobalProvider } from "@Main/view/layouts";
import { errorPageStyle } from "./ErrorPage.Style";

// Types.
export interface IErrorPageOwnProps {}

export interface IErrorPageInjectedProps extends WithSheet<typeof errorPageStyle>, IRouterContext {}

export interface IErrorPageProps extends IErrorPageOwnProps, IErrorPageInjectedProps {}

@Wrapped(GlobalProvider)
@Styled(errorPageStyle)
@Consume(routerContextManager)
export class ErrorPage extends PureComponent<IErrorPageProps> {

  public render(): ReactNode {

    const { classes } = this.props;

    return (
      <>

        <MainHeader {...{} as IMainHeaderInjectedProps}/>

        <main className={classes.content}>

          <div className={classes.label}>
            Page not found.
          </div>

          <button onClick={this.onHomeNavigated}>
            Home
          </button>

        </main>

      </>
    );
  }

  @Bind()
  private onHomeNavigated(): void {

    const { routingActions } = this.props;

    routingActions.push("/home");
  }

}
