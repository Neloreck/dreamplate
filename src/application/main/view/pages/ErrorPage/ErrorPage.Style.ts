import { IApplicationTheme } from "@Main/data/store/theme/ThemeTypes";

export const errorPageStyle = ({ palette }: IApplicationTheme) => ({
  content: {
    alignItems: "center",
    backgroundSize: "cover",
    display: "flex",
    flexDirection: "column",
    flexGrow: 1,
    flexWrap: "nowrap",
    justifyContent: "center",
    overflowY: "auto",
    width: "100%"
  },
  root: {
    backgroundColor: palette.background.paper,
    display: "flex",
    flexDirection: "column",
    height: "100%",
    width: "100%",
    transitionDuration: "250ms"
  }
});
