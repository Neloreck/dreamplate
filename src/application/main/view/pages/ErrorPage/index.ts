export { ErrorPage, IErrorPageProps } from "./ErrorPage.component";
