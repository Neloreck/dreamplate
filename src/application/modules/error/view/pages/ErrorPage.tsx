import { useManager } from "dreamstate";
import { ReactElement } from "react";
import { createUseStyles } from "react-jss";
import { Link } from "react-router-dom";

import { RouterManager } from "@/core/data/store";
import { MainHeader } from "@/core/view/components";
import { IApplicationTheme } from "@/lib/theme";

export const useStyles = createUseStyles(({ palette }: IApplicationTheme) => ({
  content: {
    alignItems: "center",
    alignSelf: "normal",
    backgroundColor: palette.background.default,
    backgroundSize: "cover",
    display: "flex",
    flexDirection: "column",
    flexGrow: 1,
    flexWrap: "nowrap",
    justifyContent: "center",
    overflowY: "auto",
    transitionDuration: "500ms",
  },
  labelCard: {
    backgroundColor: palette.secondary.light,
  },
}));

export function ErrorPage({
  classes: { content, labelCard } = useStyles(),
  routerContext: { history } = useManager(RouterManager),
}): ReactElement {
  return (
    <>
      <MainHeader />

      <main className={content}>
        <div className={labelCard}>
          {IS_DEV ? "Page was not found or current path is not serving by dev server." : "Page was not found."}
        </div>

        <Link to={history.createHref("/home")} reloadDocument>
          <button> Go Home </button>
        </Link>
      </main>
    </>
  );
}
