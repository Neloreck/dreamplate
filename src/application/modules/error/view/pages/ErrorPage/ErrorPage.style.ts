/**
 * @module @application/error
 */

import { createUseStyles } from "react-jss";

// Lib.
import { IApplicationTheme } from "@Lib/theme";

export const useStyles = createUseStyles(({ palette }: IApplicationTheme) => ({
  content: {
    alignItems: "center",
    alignSelf: "normal",
    backgroundColor: palette.background.default,
    backgroundSize: "cover",
    display: "flex",
    flexDirection: "column",
    flexGrow: 1,
    flexWrap: "nowrap",
    justifyContent: "center",
    overflowY: "auto",
    transitionDuration: "500ms"
  },
  labelCard: {
    backgroundColor: palette.secondary.light
  }
}));
