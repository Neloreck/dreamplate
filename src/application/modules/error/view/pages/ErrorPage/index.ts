/**
 * @packageDocumentation
 * @module @application/error
 */

export * from "./ErrorPage.component";
