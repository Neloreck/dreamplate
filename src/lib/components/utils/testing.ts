/**
 * @module @lib/components
 */

import { CSSResult } from "lit-element";

export const nestedShadowOf = (component: { styles: CSSResult }, ...nested: Array<string>) => {

  let commented: string = "";

  nested.forEach((it: string) => commented += `<!----><!---->${it}<!----><!---->`);

  return `${commented}<style>${component.styles}</style>`;
};
