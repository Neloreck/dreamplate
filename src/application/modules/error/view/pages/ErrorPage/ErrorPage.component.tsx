/**
 * @packageDocumentation
 * @module @application/error
 */

import { useManager } from "dreamstate";
import { ReactElement, useCallback } from "react";

// Data.
import { RouterContextManager } from "@Main/data/store";

// View.
import { MainHeader } from "@Main/view/components/MainHeader";
import { useStyles } from "./ErrorPage.style";

import "@Lib/components/custom/CustomButton";
import "@Lib/components/custom/CustomCard";

export function ErrorPage({
  classes: { content, labelCard } = useStyles(),
  routerContext: { routingActions: { hardPush } } = useManager(RouterContextManager)
}): ReactElement {

  const onHomeNavigated = useCallback(() => hardPush("/home"), []);

  return (
    <>

      <MainHeader/>

      <main className={content}>

        <custom-card class={labelCard}>
          Page was not found.
        </custom-card>

        <custom-button onClick={onHomeNavigated}>
          Go Home
        </custom-button>

      </main>

    </>
  );
}
