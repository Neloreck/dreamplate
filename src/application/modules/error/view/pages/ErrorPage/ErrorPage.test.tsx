
import { shallow } from "enzyme";

import { createMockClasses, createMockContext } from "@Lib/testing";

import { ErrorPage } from "./ErrorPage.component";

describe("Error page component rendering.", () => {
  const classes: Record<string, string> = createMockClasses();

  it("Should match snapshot.", () => {
    const wrapper = shallow(<ErrorPage classes={classes} routerContext={createMockContext()}/>);

    expect(wrapper).toMatchSnapshot("Default HOC layout.");
  });
});
