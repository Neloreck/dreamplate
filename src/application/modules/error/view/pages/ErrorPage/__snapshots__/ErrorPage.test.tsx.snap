// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error page component rendering. Should match snapshot.: Default HOC layout. 1`] = `
<Fragment>
  <MainHeader />
  <main
    className="content"
  >
    <div
      className="labelCard"
    >
      Page was not found.
    </div>
    <button
      onClick={[Function]}
    >
       Go Home 
    </button>
  </main>
</Fragment>
`;
